web/src/portico/signup.ts

```diff
@@ -15,18 +16,29 @@ $(() => {
     if ($password_field.length > 0) {
         $.validator.addMethod(
             "password_strength",
-            (value: string) => password_quality(value, undefined, $password_field),
+            (value: string) => password_quality(value, $("#pw_strength .bar"), $password_field),
             () => password_warning($password_field.val()!, $password_field),
         );
         // Reset the state of the password strength bar if the page
         // was just reloaded due to a validation failure on the backend.
         password_quality($password_field.val()!, $("#pw_strength .bar"), $password_field);
-
-        $password_field.on("input", function () {
-            // Update the password strength bar even if we aren't validating
-            // the field yet.
-            password_quality($(this).val()!, $("#pw_strength .bar"), $(this));
-        });
+        // Update the password strength bar even if we aren't validating
+        // the field yet.
+        $password_field.on(
+            "input",
+            _.debounce((e) => {
+                if (e.target.value.length === 0) {
+                    // jquery validation doesn't work for empty input
+                    password_quality(
+                        $password_field.val()!,
+                        $("#pw_strength .bar"),
+                        $password_field,
+                    );
+                } else {
+                    $password_field.valid();
```

Comment:
This feels like we are unneccessarily using jquery-validation here. Would directly calling `password_quality` cover all the validation that is done by calling `$password_field.valid().`

Reply:
Yeah, I was thinking if we can just remove the if condition and directly call `password_quality` here and keep using the `jquery-validation` for other things for now.

Reply:
That would cause the error messages to not appear, since they are handled by jquery. The password strength bar will update as expected.